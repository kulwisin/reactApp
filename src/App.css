.App {
  padding: 20px;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.tableDetails{
  width: 900px;
  height: 400px;
  overflow: scroll;
  border: 2px solid gray;
  margin-top: 12px;
  float: left;
}

.tableIndex{
  height: 200px;
  width: 890px;
  overflow: scroll;
  padding-top: 10px;
}

table, th, td {
  border: 2px solid black;
  border-collapse: collapse;
}

th, td {
  border-color: rgb(25, 36, 36);
  width: 20%;
  text-align: center;
  padding: 5px;
}

.userDetail{
  float: left;  
  width: 550px;
  height: 390px;
  border: 2px solid rgb(187, 185, 185);
  padding: 6px;
  background-color: rgb(194, 195, 195);
  margin: 12px 40px -1px 40px;
}

::-webkit-scrollbar {
  -webkit-appearance: none;
  width: 10px;
}

::-webkit-scrollbar-thumb {
  border-radius: 5px;
  background-color: rgba(0,0,0,.5);
  -webkit-box-shadow: 0 0 1px rgba(255,255,255,.5);
}

.userDetailsSection{
  padding: 4px;
}

.recordLength{
  display: inline-block;
}