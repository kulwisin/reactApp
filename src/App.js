import React, {useRef, useEffect, useState} from 'react';
import axios from 'axios';
import UserDataTable from './userDataTable';
import './App.css';

export default function App() {
  const [myState, setState] = useState([]);
  const previousValue = useRef(null);
  let timerID;

  const getData = async () => {
    await axios.get("https://randomuser.me/api/?results=100").then((response) =>{
      setState(response.data.results)
      previousValue.current = response.data.results;
    })
  }
  
  useEffect(() => {
    getData()
  }, [])

  const handleSearch = (event) => {
    clearTimeout(timerID);
    timerID = setTimeout(() =>{
      const newState = Object.values(previousValue.current).filter(item => item.name.first.toLowerCase().includes(event.target.value.toLowerCase()));
      setState(newState);
    }, 1000)
  };

  return (
    <div className="App">
      <div>
        <label htmlFor="search">
          Search by Name : 
          <input id="search" placeholder="Type..." type="text" onChange={handleSearch} />
        </label>
        <UserDataTable {...myState}/>
        <div className={"recordLength"}>
          <p>Records {myState.length} </p>
        </div>
      </div>
    </div>
  );
};