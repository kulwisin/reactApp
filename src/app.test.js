import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('Information App', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});