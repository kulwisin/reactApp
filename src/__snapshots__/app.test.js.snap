// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information App 1`] = `
<div
  className="App"
>
  <div>
    <label
      htmlFor="search"
    >
      Search by Name :
      <input
        id="search"
        onChange={[Function]}
        placeholder="Type..."
        type="text"
      />
    </label>
    <div>
      <div
        className="tableDetails"
      >
        <table
          className="tableIndex"
          data={Object {}}
        >
          <tr>
            <th>
              Name
            </th>
            <th>
              Gender
            </th>
            <th>
              Emial
            </th>
            <th>
              Phone
            </th>
            <th>
              More Info
            </th>
          </tr>
        </table>
      </div>
      <div
        className="userDetail"
      >
        <h2>
          User Details Section
        </h2>
      </div>
    </div>
    <div
      className="recordLength"
    >
      <p>
        Records 
        0
         
      </p>
    </div>
  </div>
</div>
`;
